packages/blazor-workspace/NimbleBlazor/wwwroot/NimbleBlazor.lib.module.js

```diff
@@ -5,169 +5,204 @@
  * Register the custom event types used by Nimble components.
  *
  * JavaScript initializer for NimbleBlazor project, see
- * https://docs.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/?view=aspnetcore-6.0#javascript-initializers
+ * https://learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/?view=aspnetcore-8.0 and
+ * https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/startup?view=aspnetcore-8.0
  */
 
-export function afterStarted(Blazor) {
-    if (window.NimbleBlazor.calledAfterStarted) {
-        console.warn('Attempted to initialize Nimble Blazor multiple times!'); // eslint-disable-line
-        return;
-    }
+const initializer = (function () {
```

Comment:
In an es6 module, variables that are defined aren't global, so this IIFE is unnecessary and not common practice in our other source so I would avoid it. Creating top level bindings and referencing them is not global mutable state, it's local to the module:

```ts
let blah;
export updatetBlah = () => {blah = 7;}
```

The concern is exposing global mutable properties.

```
window.Blazor.ready = true;
```

Can we instead export a function to check the state?

```
let isReady;

window.Blazor.isReady = () => isReady;
```

Reply:
Makes sense.

Removed IIFE. Replaced isInitialized boolean with an isReady() function.